src/main/java/info/fingo/urlopia/history/HistoryLogService.java

```diff
@@ -134,11 +135,29 @@ public void create(HistoryLogInput historyLog,
                             " because decider with id: {} does not exist"), targetUserId, deciderId);
                     return NoSuchUserException.invalidId();
                 });
-        var prevHistoryLog = historyLogRepository.findFirstByUserIdOrderByIdDesc(targetUserId);
+        create(historyLog, targetUser, decider);
+    }
+
+    public void createBySystem(HistoryLogInput historyLog,
+                               Long targetUserId) {
+        var targetUser = userRepository
+                .findById(targetUserId)
+                .orElseThrow(() -> {
+                    log.error("Could not create new history log for nonexistent user with id: {}", targetUserId);
+                    return NoSuchUserException.invalidId();
+                });
+        create(historyLog, targetUser, null);
```

Comment:
Instead of passing `null` consider adding a new method that will explicitly state what is going on here